datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Domain/SetDomainModal.tsx

```diff
@@ -87,56 +99,66 @@ export const SetDomainModal = ({ visible, onClose, refetch }: Props) => {
         }
     };
 
-    const handleSearch = (text: string) => {
-        if (text.length > 2) {
-            domainSearch({
+    const onOk = async () => {
+        if (!selectedDomain) {
+            return;
+        }
+        try {
+            await setDomainMutation({
                 variables: {
-                    input: {
-                        type: EntityType.Domain,
-                        query: text,
-                        start: 0,
-                        count: 5,
-                    },
+                    entityUrn: urn,
+                    domainUrn: selectedDomain.urn,
                 },
             });
+            message.success({ content: 'Updated Domain!', duration: 2 });
+        } catch (e: unknown) {
+            message.destroy();
+            if (e instanceof Error) {
+                message.error({ content: `Failed to set Domain: \n ${e.message || ''}`, duration: 3 });
+            }
         }
+        setSelectedDomain(undefined);
+        refetch?.();
+        onCloseModal();
+    };
+
+    const tagRender = (props) => {
+        // eslint-disable-next-line react/prop-types
+        const { label, closable, onClose } = props;
+        const onPreventMouseDown = (event) => {
+            event.preventDefault();
+            event.stopPropagation();
+        };
+        return (
+            <StyleTag onMouseDown={onPreventMouseDown} closable={closable} onClose={onClose}>
+                {label}
+            </StyleTag>
+        );
     };
 
+    const selectValue = (selectedDomain && [selectedDomain?.displayName]) || [];
+
     // Handle the Enter press
     useEnterKeyListener({
         querySelectorToExecuteClick: '#setDomainButton',
     });
 
-    const renderSearchResult = (result: SearchResult) => {
-        const displayName = entityRegistry.getDisplayName(result.entity.type, result.entity);
-        return (
-            <SearchResultContainer>
-                <SearchResultContent>
-                    <SearchResultDisplayName>
-                        <div>{displayName}</div>
-                    </SearchResultDisplayName>
-                </SearchResultContent>
-                <Link
-                    target="_blank"
-                    rel="noopener noreferrer"
-                    to={() => `/${entityRegistry.getPathName(result.entity.type)}/${result.entity.urn}`}
-                >
-                    View
-                </Link>{' '}
-            </SearchResultContainer>
-        );
-    };
+    function clearInput() {
+        setInputValue('');
+    }
 
-    const selectValue = (selectedDomain && [selectedDomain?.displayName]) || [];
+    function handleBlur() {
```

Comment:
Does this mean that when we unfocus the text box is cleared? We probably don't want this!

Reply:
Checked and fixed.